### PR TITLE
git7234 increase scope  for unique properties check

### PR DESCRIPTION
**Pull request by davidradl**
Signed-off-by: David Radley <david_radley@uk.ibm.com>

The generic handler code that checks for uniqueness for unique attributes has been changed to check at the level of the type where the unique attribute is defined , rather than the type in the entity that was supplied. 

## Description

The generic handler code that checks for uniqueness for unique attributes has been changed to check at the level of the type where the unique attribute is defined , rather than the type in the entity that was supplied. 

The code starts with the type supplied in the entity and looks for the unique property in it, it checks the super types until it finds th property, then looks for existing entities using that type.

## Testing

I have tested by creating a glossary through the subject area omas with a qualified name then created an asset with the same qualified name, this fails.  
## Release Notes & Documentation

Yes this is important as this could effect existing use cases.
## Additional notes

This effects create and update.

### CODE REVIEW

**Comment by davidradl** [comment]
@lpalashevski here is the pr for #7234 - you expressed some interest in it.

**Review by mandy-chessell — APPROVED** [review]
Looks good - one small code layout problem.

File: open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/OpenMetadataAPIGenericHandler.java [hunk 1]
```diff
@@ -3236,14 +3236,46 @@ public void validateUniqueProperty(String entityGUID,
...
+                    }
+                }
+            }
+
+            TypeDefLink superTypeDefLink = potentialOwningEntityTypeDef.getSuperType();
+
+            if (superTypeDefLink != null)
+            {
+                entityTypeNameToCheck = superTypeDefLink.getName();
+            } else
```
[the remark this answers is not in the document]
Comment (davidradl) [reply open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/OpenMetadataAPIGenericHandler.java:3264]: @mandy-chessell sorry to be a pain - what do I need to change here . I can't see it.  Is it the comment style?

[the remark this answers is not in the document]
Comment (mandy-chessell) [reply open-metadata-implementation/common-services/generic-handlers/src/main/java/org/odpi/openmetadata/commonservices/generichandlers/OpenMetadataAPIGenericHandler.java:3264]: The else is on the wrong line

**Pull request merged**